src/main/scala/org/broadinstitute/dsde/firecloud/model/ModelJsonProtocol.scala

```diff
@@ -131,21 +140,71 @@ object ModelJsonProtocol {
         case None => None
         case _ => throw DeserializationException("unexpected json type for " + SIZE)
       }
-      LibrarySearchParams(term, from, size)
+      val fields: Map[String, Seq[String]] = data.getOrElse(SEARCH_FIELDS, None) match {
+        case some:JsObject => some.convertTo[Map[String,Seq[String]]]
+        case None => Map.empty
+      }
+      val aggs = data.getOrElse(FIELD_AGGREGATIONS, None) match {
+        case some: JsArray => some.convertTo[Seq[String]]
+        case None => Seq.empty
+        case _ => throw DeserializationException("unexpected json type")
+      }
+      val maxAggs = data.getOrElse(MAX_AGGREGATIONS, None) match {
+        case JsNumber(x) => Some(x.intValue)
+        case None => None
+        case _ => throw DeserializationException("unexpected json type")
+      }
+      LibrarySearchParams(term, fields, aggs, maxAggs, from, size)
+    }
+  }
```

Comment:
this is a lot of boilerplate, but I'm struggling for a better way to do it. I'll keep thinking on it, but there may not be a super elegant way.